backend/src/api/workspaces.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express')
+const { Workspace } = require('../models')
+const { directorsOnly, errorWrap } = require('../middleware')
+const router = express.Router()
+
+// Get all workspaces
+// TODO: when authentication server is integrated, only show the workspace that
+// belong to the caller of this endpoint.
+router.get(
+  '/',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const workspaces = await Workspace.find()
+    res.json({
+      code: 200,
+      result: workspaces,
+      success: true
+    })
+  })
+)
+
+// Find workspace by name
+// TODO: when authentication server is integrated, only show the workspace if it
+// belongs to the caller of this endpoint.
+router.get(
+  '/:workspaceName',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const workspaceName = req.params.workspaceName
+
+    if (!workspaceName) {
+      return res.json({
+        code: 400,
+        message: 'malformed request',
+        success: false
+      })
+    }
+
+    const workspace = await Workspace.find({ name: workspaceName })
+
+    res.json({
+      code: 200,
+      result: workspace,
+      success: true
+    })
+  })
+)
+
+// Create a new workspace
+router.post(
+  '/',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const owner = req.body.owner
+    const workspaceName = req.body.name
+    if (!owner || !workspaceName) {
+      return res.json({
+        code: 400,
+        message: owner ? 'Invalid owner name' : 'Invalid workspace name',
+        success: false
+      })
+    }
+
+    const workspace = new Workspace({
+      name: workspaceName,
+      owner
+    })
+    await workspace.save()
+
+    res.json({
+      code: 200,
+      message: 'Successfully created workspace',
+      success: true
+    })
+  })
+)
+
+// Transfer ownership of a workspace to someone else
+router.put(
+  '/transfer/:workspaceName',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const newOwner = req.body.owner
+    const name = req.params.workspaceName
```

Comment:
maybe rename `name` bc it can be confusing to have both an owner and a workspace name